src/PointwiseFunctions/InitialDataUtilities/InitialData.hpp

```diff
@@ -18,7 +18,7 @@ namespace initial_data {
  * \brief The abstract base class for initial data of evolution systems. All
  * analytic solutions and analytic data must virtually inherit from this class.
  */
-class InitialData : public PUP::able {
+class InitialData : public virtual PUP::able {
```

Comment:
@nilsdeppe Here's the virtual inheritance again that I dropped from another PR. It's needed here to derive the `TovStar` class from both `AnalyticSolution` and `RelEuler::TovStar`. Do you think this is fine, or do you have another suggestion how to handle this?

Reply:
Looking into this a bit for the evolution code, I think it's fine to do virtual inheritance with serialization as far as I can tell and reason about. However, I don't need this class to inherit virtually, I have things inherit virtually from `evolution::initial_data::InitialData`. Once your at Caltech we can sit down together so I can understand where the difference is arising.